sap/adt/dataelement.py

```diff
@@ -0,0 +1,221 @@
+            self._metadata.data_element.data_type = ''
+            self._metadata.data_element.data_type_length = '0'
+            self._metadata.data_element.data_type_decimals = '0'
+        if self._metadata.data_element.type == 'predefinedAbapType':
+            self._metadata.data_element.type_name = ''
+
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_short_length = self._metadata.data_element.label_short_length if not self._metadata.data_element.label_short_length else LABELS_LENGTH['short']
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_medium_length = self._metadata.data_element.label_medium_length if not self._metadata.data_element.label_medium_length else LABELS_LENGTH['medium']
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_long_length = self._metadata.data_element.label_long_length if not self._metadata.data_element.label_long_length else LABELS_LENGTH['long']
+        # pylint: disable=line-too-long
+        self._metadata.data_element.label_heading_length = self._metadata.data_element.label_heading_length if not self._metadata.data_element.label_heading_length else LABELS_LENGTH['heading']
+
+        # pylint: disable=line-too-long
+        self._metadata.data_element.deactivate_input_history = self._metadata.data_element.deactivate_input_history if self._metadata.data_element.deactivate_input_history is not None else False
+        # pylint: disable=line-too-long
+        self._metadata.data_element.change_document = self._metadata.data_element.change_document if self._metadata.data_element.change_document is not None else False
+        # pylint: disable=line-too-long
+        self._metadata.data_element.left_to_right_direction = self._metadata.data_element.left_to_right_direction if self._metadata.data_element.left_to_right_direction is not None else False
+        # pylint: disable=line-too-long
+        self._metadata.data_element.deactivate_bidi_filtering = self._metadata.data_element.deactivate_bidi_filtering if self._metadata.data_element.deactivate_bidi_filtering is not None else False
+
+    def validate(self):
+        """Validate Data Element setup"""
+
+        if self._metadata.data_element.type == 'domain' and not self._metadata.data_element.type_name:
+            raise SAPCliError('Domain name must be provided (--domain_name) if the type (--type) is "domain"')
+        if self._metadata.data_element.type == 'predefinedAbapType' and not self._metadata.data_element.data_type:
+            # pylint: disable=line-too-long
+            raise SAPCliError('Data type name must be provided (--data_type) if the type (--type) is "predefinedAbapType"')
+
+    @property
+    def text(self):
+        """ Downloads text representation of the SAP Object """
+
+        return self._connection.get_text(f'{self.uri}', {'Accept': 'application/vnd.sap.adt.dataelements.v2+xml'})
```

Comment:
Doesn't this return the unparsed XML representation?

Reply:
I wanted to say, if data element does not have "text/plain" reprezentation, do try to provide it as XML.

Reply:
Removed in new code version since it is not needed any more